### ISSUE TITLE
[Feature Request] Connecting melonds to a real ds

### ISSUE THREAD

**Issue by EpicDoodleboy2** [melonDS-emu/melonDS#1554]
This feature that I've requested might be useful for people that who might want to play with someone that has a pc but they don't own one but do have a ds so that might work out   or people who just want to play New Super Mario Bros DS Co-Op and doesn't want to share window to window via idk parsec

**Comment by nadiaholmquist** [comment]
While this would be really cool, it's a very hard or maybe impossible problem to solve. [Here's a pretty good explanation of why that is](https://www.reddit.com/r/emulation/comments/ylg8hk/comment/iuz6kbi/?utm_source=reddit&utm_medium=web2x&context=3).

**Comment by EpicDoodleboy2** [comment]
I was thinking melonds could use one of the pcs antennas because there is two, wifi and one for searching for devices if we use the second antenna to search for a ds it might work

**Comment by EpicDoodleboy2** [comment]
but i'll agree it will be very hard

**Comment by Real96** [comment]
> This feature that I've requested might be useful for people that who might want to play with someone that has a pc but they don't own one but do have a ds so that might work out or people who just want to play New Super Mario Bros DS Co-Op and doesn't want to share window to window via idk parsec

I was able to do this with altwfc. I made a script to create your own server for this

**Comment by EpicDoodleboy2** [comment]
> > This feature that I've requested might be useful for people that who might want to play with someone that has a pc but they don't own one but do have a ds so that might work out or people who just want to play New Super Mario Bros DS Co-Op and doesn't want to share window to window via idk parsec
> 
> I was able to do this with altwfc, melonds vs real console on Mario Kart. I made a script to create your own server for this

Hey Real96, MKDS Is an Online compatible game I was thinking of a local multiplayer game like sonic rush, sm64ds and nsmb co-op
transmitting a signal with the right strength and would show on download play or smth , also I use Wiimmfi not altwfc

**Comment by Real96** [comment]
> > > This feature that I've requested might be useful for people that who might want to play with someone that has a pc but they don't own one but do have a ds so that might work out or people who just want to play New Super Mario Bros DS Co-Op and doesn't want to share window to window via idk parsec
> > 
> > 
> > I was able to do this with altwfc, melonds vs real console on Mario Kart. I made a script to create your own server for this
> 
> Hey Real96, MKDS Is an Online compatible game I was thinking of a local multiplayer game like sonic rush, sm64ds and nsmb co-op transmitting a signal with the right strength and would show on download play or smth , also I use Wiimmfi not altwfc was it download play settings or wfc settings or i'm being stupid

Yeah, it is online playing. At least we have a way to do this on some games. Locally or not locally, even if it is through wifi

**Comment by EpicDoodleboy2** [comment]
I've actually tried to go online with real ds and melonds but I couldn't get it to connect to each other even with friend codes or a WW match

**Comment by Real96** [comment]
> I've actually tried to go online with real ds and melonds but I couldn't get it to connect to each other even with friend codes or a WW match

I just connected the real ds and melonDS to my local dwc server, no patch needed. They could find without problems and i was able to do a run with Mario Kart. No friend code needed.

**Comment by EpicDoodleboy2** [comment]
Hold on are you using TWL menu++ on a DSI with forced updated wi-fi settings?

**Comment by Real96** [comment]
Old DS, the fat one. Original Mario kart, no patch. Just the server with ssl

**Comment by EpicDoodleboy2** [comment]
I've tried a real cart before, it didn't work but then again I'm using wiimmfi with a mobile hotspot with no security but idk if wiimmfi has ssl

**Comment by Real96** [comment]
> I've tried a real cart before, it didn't work but then again I'm using wiimmfi with a mobile hotspot with no security but idk if wiimmfi has ssl
> 
> --- tutorial: https://www.youtube.com/watch?v=Sh0AZR-tKwM

I think you can connect to it without patching the game, so wiimmfi should have ssl set up too.
In case you want to give it a try, this is the script [link](https://github.com/Real96/Nintendo-DWC-Installer-Script)

**Comment by Jdbye** [comment]
I think it's possible, but might require specific hardware. The wifi multiboot server that sends demo roms and the wifime exploit using DS Download Play was always limited to specific (and ancient) wifi chipsets. 
If at all possible, this would still be a cool feature to have though. I still have a Nintendo WFC USB Connector laying around, which has the right chipset. And I bet many others do as well.
Most of us probably don't carry a DS around these days, but we have smartphones, Steam Decks, laptops and what have you, and being able to use local wifi the same way we could on real hardware back in the day would be cool.
